scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1388,6 +1388,9 @@
       (.setName metadata pod-name)
       (.setNamespace metadata namespace)
       (.setLabels metadata labels)
+      ; Only include finalizers with real pods.
+      (when-not (synthetic-pod? pod-name)
+        (.setFinalizers metadata (list RemoveFinalizerHelper/collectResultsFinalizer)))
```

Comment:
Maybe we should put this behind a feature flag

Reply:
I don't like those in general, if you're just going to leave it permanently enabled, (and the system will behave incorrectly with it disabled) why make it configurable?

Reply:
Agreed, we'd want to follow-up with a PR to remove the feature flag once we see it working as expected for a few days in production.

Reply:
Agreed